### ISSUE TITLE
Packages not updated, sometimes...

### ISSUE THREAD

**Issue by powellchristoph** [mardiros/pyshop#47]
I have repeated experienced an issue where PyPi will release a new version of a package and pyshop is not updating properly. 

I submitted a patch #46 that fixes part of this, but I am currently unable to located the source of the issue.

Current problem with [Pecan](https://pypi.python.org/pypi/pecan):

They released a new version 0.8.3 on 1/12/2015. You can see from the logs below that it recognized refresh interval (24 hours) and issued a refresh on the package. But for some reason, it did not find the new version and cache it.
```
2015-01-12 19:52:47,675 INFO  [pyshop.security][Dummy-4] [104.130.1.123] GET http://wheel.cldbckp.com/test/simple/pecan/
28303 2015-01-12 19:52:47,854 INFO  [pyshop.views.base][Dummy-4] dispatch view Show
28304 2015-01-12 19:52:47,855 DEBUG [pyshop.views.simple][Dummy-4] "2 days, 21:12:23.652369" > "1 day, 0:00:00": True
28305 2015-01-12 19:52:47,855 INFO  [pyshop.views.simple][Dummy-4] refresh package pecan
28306 2015-01-12 19:52:47,954 INFO  [pyshop.views.simple][Dummy-4] package pecan mirrored
28307 2015-01-12 19:52:48,281 INFO  [pyshop.security][Dummy-3] [104.130.1.123] GET http://wheel.cldbckp.com/test/simple/pecan/
28308 2015-01-12 19:52:48,458 INFO  [pyshop.views.base][Dummy-3] dispatch view Show
28309 2015-01-12 19:52:48,460 DEBUG [pyshop.views.simple][Dummy-3] "0:00:00.779265" > "1 day, 0:00:00": False
28310 2015-01-12 19:52:48,464 INFO  [pyshop.views.simple][Dummy-3] package pecan mirrored
```
If I check manually with pyshop_shell, I can see that the pypi.proxy finds the package and lists it.
```
(test)root@bus1a-cbo-iad:/opt/pyshop/test# pyshop_shell test.ini
Welcome to PyShop shell.
In [1]: pkg = Package.by_name(session, 'pecan')

In [2]: pkg.versions
Out[2]:
[u'0.8.2',
 u'0.8.1',
 u'0.8.0',
 u'0.7.0',
 u'0.6.1',
 u'0.6.0',
 u'0.5.0',
 u'0.4.5',
 u'0.4.4',
 u'0.4.3',
 u'0.4.2',
 u'0.4.1',
 u'0.3.2',
 u'0.3.1',
 u'0.3.0',
 u'0.2.4',
 u'0.2.3',
 u'0.2.2',
 u'0.2.1',
 u'0.2.0',
 u'0.2.0b2',
 u'0.2.0a']

In [3]: from pyshop.helpers import pypi

In [4]: pypi.proxy.package_releases('pecan', True)
Out[4]:
['0.8.3',
 '0.8.2',
 '0.8.1',
 '0.8.0',
 '0.7.0',
 '0.6.1',
 '0.6.0',
 '0.5.0',
 '0.4.5',
 '0.4.4',
 '0.4.3',
 '0.4.2',
 '0.4.1',
 '0.3.2',
 '0.3.1',
 '0.3.0',
 '0.2.4',
 '0.2.3',
 '0.2.2',
 '0.2.1',
 '0.2.0',
 '0.2.0b2',
 '0.2.0a']
```
When I force a refresh by using the refresh button, it pulls it in perfectly.
```
2015-01-14 18:33:00,290 INFO  [pyshop.views.base][Dummy-4] dispatch view Show
3307995 2015-01-14 18:33:00,291 INFO  [pyshop.views.simple][Dummy-4] refresh package pecan
3307996 2015-01-14 18:33:00,338 INFO  [pyshop.views.simple][Dummy-4] Create release 0.8.3 for package pecan
3307997 2015-01-14 18:33:00,339 INFO  [pyshop.views.simple][Dummy-4] Looking for author Jonathan LaCour
3307998 2015-01-14 18:33:01,486 INFO  [pyshop.views.simple][Dummy-4] package pecan mirrored
```

**Comment by mardiros** [comment]
Why do you think that patch #46 did not fix it completly ?

**Comment by powellchristoph** [comment]
Because of here:
```
DEBUG [pyshop.views.simple][Dummy-4] "2 days, 21:12:23.652369" > "1 day, 0:00:00": True
INFO  [pyshop.views.simple][Dummy-4] refresh package pecan
INFO  [pyshop.views.simple][Dummy-4] package pecan mirrored
```
It recognized the time-delta as 2 days which is greater than the configured 1 day and it still did not refresh the package. #46 address the *pkg.update_at* getting reset for each pip request.

I will keep watching it. I am running three pyshop instances which are extremely active so I should notice a problem if something isn't refreshed.

**Comment by mardiros** [comment]
Ok I made this change ^, and for what I test, it works...
I should also test before make that change :)

**Comment by mardiros** [comment]
Reopen me if its not fixed in pyshop 1.2.2

**Comment by powellchristoph** [comment]
Sounds great. I will schedule an update soon. Thanks!
